---
title: "A Mediation Approach to Discovering Causal Relationships Between the Metabolome and DNA Methylation in Type 1 Diabetes"
date: ""
author:
- Timothy Vigers, B.A.^[Department of Biostatistics and Informatics, Colorado School of Public Health, Aurora, Colorado, USA], Lauren A. Vanderlinden, M.S.^[Department of Biostatistics and Informatics, Colorado School of Public Health, Aurora, Colorado, USA], Randi K. Johnson, Ph.D.^[University of Colorado School of Medicine, Division of Biomedical Informatics and Personalized Medicine], Patrick M. Carry, M.S., Ivana Yang, Ph.D.^[University of Colorado School of Medicine, Division of Biomedical Informatics and Personalized Medicine], Brian C. DeFelice, Ph.D., Alexander M. Kaizer, Ph.D.^[Department of Biostatistics and Informatics, Colorado School of Public Health, Aurora, Colorado, USA], Marian Rewers, M.D., Ph.D., Oliver Fiehn, Ph.D., Jill M. Norris, Ph.D., Katerina Kechris, Ph.D.^[Department of Biostatistics and Informatics, Colorado School of Public Health, Aurora, Colorado, USA]
bibliography: "/Volumes/Vault/Miscellaneous/references.bib"
csl: "/Volumes/Vault/Miscellaneous/american-medical-association.csl"
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,dpi = 600)
knitr::opts_knit$set(root.dir = "~/Dropbox/School/MS Thesis")
library(rjson)
library(org.Hs.eg.db)
library(annotate)
library(DiagrammeRsvg)
library(rsvg)
library(GeneBook)
library(arsenal)
library(DiagrammeR)
library(broom)
library(data.table)
library(knitr)
library(regmedint)
library(mediation)
library(dplyr)
library(tidyr)
library(pheatmap)
library(ggplot2)
options(knitr.kable.NA = '')
```

```{r}
# Results data
load("./data/raw_data/psv_sv_dataset.Rdata")
load("./data/mediation/methyl_psv_results.Rdata")
load("./data/mediation/methyl_psv_candidates_p_01.Rdata")
load("./data/mediation/metab_psv_candidates_p_01.Rdata")
load("./data/raw_data/all_metabolite_anno.RData")
load("./data/raw_data/annotation.850K.Rdata")
anno = as.data.frame(anno)
unknowns = read.csv("./data/raw_data/unknown_metabs.csv")
unknowns = unknowns$metab
```

# Abstract

Environmental factors including viruses, diet, and the metabolome have been linked with the appearance of islet autoimmunity (IA) that precedes development of type 1 diabetes (T1D). We measured global DNA methylation (DNAm) and untargeted metabolomics prior to IA and at the time of seroconversion to IA in 92 IA cases and 91 controls from the Diabetes Autoimmunity Study in the Young (DAISY) study. Causal mediation models were used to identify seven DNAm probe-metabolite pairs in which the metabolite measured at IA mediated the protective effect of the DNAm probe measured prior to IA against IA risk. These pairs included five DNAm probes mediated by histidine (a metabolite known to affect T1D risk), one probe (cg01604946) mediated by phostidyl choline p-32:0 or o-32:1, and one probe (cg00390143) mediated by sphingomyelin d34:2. The top 100 DNAm probes were over-represented in six reactome pathways at the FDR < 0.1 level (q = 0.071), including transport of small molecules and inositol phosphate metabolism. While the causal pathways in our mediation models require further investigation to better understand the biological mechanisms, we identified seven methylation sites that may improve our understanding of epigenetic protection against T1D as mediated by the metabolome.

\newpage

# Introduction

Type 1 diabetes (T1D) is an autoimmune disease characterized by the production of antibodies which target pancreatic $\beta$-cells. The disease currently affects over 30 million people worldwide,[@rakyanIdentificationTypeDiabetes2011] and is increasing by 3-4% per year on average.[@tuomilehtoEmergingGlobalEpidemic2013]

Genetic predisposition accounts for some of the etiology of T1D (siblings of an individual with T1D have a relative risk 15 times higher than those without a sibling with T1D)[@polychronakosUnderstandingTypeDiabetes2011] and explains some geographic variation in incidence.[@tuomilehtoEmergingGlobalEpidemic2013] Human leukocyte antigen (HLA) genes were the first to be linked to T1D and account for much of the genetic predisposition to the disease, but genome-wide association studies (GWAS) have also identified more than 40 other T1D-associated loci.[@atkinsonTypeDiabetes2014] However, it is still unclear how these multiple loci[@thetype1diabetesgeneticsconsortiumGenomewideAssociationStudy2009] interact with one another and the environment to produce a T1D diagnosis. In addition to the complex genetics of T1D, low monozygotic (MZ) twin concordance (approximately 50%) and the increasing incidence over time[@tuomilehtoEmergingGlobalEpidemic2013] support the theory that non-genetic factors play an important role in T1D development.[@rakyanIdentificationTypeDiabetes2011] 

Epigenetic differences may be important contributors to T1D etiology. Changes in methylation have been associated with other autoimmune conditions, and monozygotic twins can be epigenetically heterogeneous despite sharing identical genetic code.[@rakyanIdentificationTypeDiabetes2011]  Rakyan et al.[@rakyanIdentificationTypeDiabetes2011] and Stefan et al.[@stefanDNAMethylationProfiles2014] performed epigenome-wide association studies (EWAS) in discordant and concordant twin pairs and found that methylation profiles were more similar among participants with T1D than to unaffected twins. Epigenetics profiles were also combined with GWAS data and differentially methylated sites were mapped to 6 well known T1D susceptibility genes, including two major histocompatibility complex (MHC) genes and several HLA loci.[@stefanDNAMethylationProfiles2014;@rakyanIdentificationTypeDiabetes2011] Finally, Johnson et al. found that T1D cases had different longitudinal methylation patterns compared to controls prior to diagnosis.[@johnsonLongitudinalDNAMethylation2020] 

Environmental factors including viruses, diet, and the metabolome have also been linked with islet autoimmunity (IA)[@oresicDysregulationLipidAmino2008] and T1D etiology.[@atkinsonTypeDiabetes2014] Previous studies have found associations between T1D and differentially expressed phospholipids and sphingolipids, excretion of modified amino acids, and vitamin D (and related compounds on its metabolic pathway).[@frohnertMetabolomicsChildhoodDiabetes2016;@atkinsonTypeDiabetes2014]

To date, the effects of the metabolome and DNA methylation have been studied separately in T1D, thus, the combined nutrigenomics of T1D remain unclear. Early **in vitro** studies confirmed metabolome-dependent alterations in DNA methylation associated with various cancers,[@chiacchieraEpigeneticMethylationsTheir2013;@putluriMetabolomicProfilingReveals2011] and even identified oncometabolites associated with the development of glioma.[@wellerIsocitrateDehydrogenaseMutations2011] More recent human studies also support the connection between the metabolome and methylation in breast cancer,[@alakwaaBioinformaticsAnalysisMetabolomics2020] colorectal cancer,[@wangMultiomicProfilingReveals2020] and smoking-related diseases.[@huangUntargetedMetabolomicsReveals2018] A 2018 study by Zaghlool et al. used linear models to link metabolomics and DNA methylation with type 2 diabetes (T2D) and obesity, and used Mendelian randomization (MR) to provide evidence for "a causal effect of metabolite levels on methylation of obesity-associated CpG sites."[@zaghloolDeepMolecularPhenotypes2018] 

Causal interpretations are of particular interest when integrating epigenetic and metabolomics data. The counterfactual framework for mediation analyses is widely used in statistics and epidemiology because it can allow for causal interpretations of observational data. Also, it provides a relatively intuitive interpretation of mediation effects by quantifying how much an outcome of interest would likely differ given a change in either the exposure or mediator.[@vanderweeleExplanationCausalInference2015] Because mediation is defined based on an assumed causal model, it is important that causality assumptions are reasonable.[@fiskeHandbookSocialPsychology2010] It is difficult to evaluate causal assumptions across the epigenome, but longitudinal study designs allow for mediation models in which the exposure variable was measured prior to the mediator. To our knowledge, this is the first study to examine causal links between DNAm, the metabolome, and T1D.

The Diabetes Autoimmunity Study in the Young (DAISY) follows 2547 high-risk children in Colorado for the development of IA and T1D. Follow-up includes blood sample collection at 9, 15, and 24 months, then annual collection until one or more autoantibodies are detected. Participants who test positive for one or more autoantibodies are asked to follow an accelerated protocol with visits and blood sample collection every 3-6 months. IA  is defined as two consecutive visits at which a confirmed autoantibody to insulin, GAD65, IA-2, or ZnT8 was detected (Figure 1).[@johnsonMetaboliterelatedDietaryPatterns2019] The children are followed over time, until they are diagnosed with T1D by a physician using American Diabetes Association criteria. Thus, DAISYâ€™s study design allows us to include exposure and mediator variables that are temporally separated in our mediation models because methylation was measured prior to seroconversion [PSV], defined as the autoantibody-negative visit preceding the visit at which the child first tested autoantibody positive [SV],  and metabolite was measured at seroconversion visit [SV] (and vice versa). This aids in interpretation of the results and in theory does not violate the assumptions of a mediation analysis.

\newpage

# Results

## Figure 1: Study Design and Candidate Selection

```{r fig.height=8,fig.width=8,eval=FALSE,include=FALSE}
flow = grViz("digraph {

graph [layout = dot, rankdir = LR]

node [shape = rectangle]

birth [label = 'Birth']
none [label = 'Abx- \n Visits']
first [label = 'First Abx+ \n Visit']
second [label = 'Second Abx+ \n Visit (IA)']
psv [label = 'PSV']
sv [label = 'SV']
t1d [label = 'T1D']

# edge definitions with the node IDs
birth -> none -> first -> second -> t1d;
{rank=same; first -> sv}
{rank=same; none -> psv}
}")
med1 = grViz("
  graph{
  node [shape = box,fontname = Helvetica]
  a[label= 'Metabolite at SV']
  b[label= 'Methylation at PSV']
  c[label= 'IA']
  a -- b [label = 'p < 0.01',fontsize = 9]
  a -- c [label = 'p< 0.01',fontsize = 9]
  {rank = same;b -- c [label = 'p< 0.01',fontsize = 9]}
  }
")
med2 = grViz("
  graph{
  node [shape = box,fontname = Helvetica]
  a[label= 'Methylation at SV']
  b[label= 'Metabolite at PSV']
  c[label= 'IA']
  d[label= 'IA']
  a -- b [label = 'p < 0.01',fontsize = 9]
  a -- c [label = 'p< 0.01',fontsize = 9]
  {rank = same;b -- c [label = 'p< 0.01',fontsize = 9]}
  }
")
flow %>% export_svg(.) %>% charToRaw(.) %>% rsvg_png("./images/flow.png")
med1 %>% export_svg(.) %>% charToRaw(.) %>% rsvg_png("./images/med1.png")
med2 %>% export_svg(.) %>% charToRaw(.) %>% rsvg_png("./images/med2.png")
```

![](/Users/timvigers/Dropbox/School/MS Thesis/images/fig1.jpg)

Figure legend:

(a) A flowchart depicting progression from birth to islet autoimmunity (IA) and the Diabetes Autoimmunity Study in the Young (DAISY) study visits. To reduce computation time, we performed simple linear regression without covariate adjustment to identify pairs of DNA methylation (DNAm) probes and metabolites for mediation analysis. Candidate pairs were significantly associated at the p < 0.01 level, and both DNAm probe and metabolite were associated with development of IA at the p < 0.01 level. (b) We examined pairs with DNAm measured pre-seroconversion (PSV) and metabolite measured at seroconversion (SV). (c) We also examined pairs with metabolite measured at PSV and DNAm measured at SV in separate analyses.

\newpage

## Figure 2: Correlation Heat Maps Between Temporally Separated Omics Data

```{r include=FALSE}
# First plots
m = cor(psv[,unique(methyl_psv_candidates[,1])],sv[,unique(methyl_psv_candidates[,2])],
        use = "pairwise.complete.obs")
methyl_psv_heat = pheatmap(m,show_rownames = F,show_colnames = F,
         main = "a")
ggsave(file="/Users/timvigers/Dropbox/School/MS Thesis/images/heatmapa.tiff",plot = methyl_psv_heat,dpi = 600)
# Second
m = cor(sv[,unique(metab_psv_candidates[,1])],psv[,unique(metab_psv_candidates[,2])],
        use = "pairwise.complete.obs")
metab_psv_heat = pheatmap(m,show_rownames = F,show_colnames = F,
         main = "b")
ggsave(file="/Users/timvigers/Dropbox/School/MS Thesis/images/heatmapb.tiff",plot = metab_psv_heat,dpi = 600)
```

Figure legend:

Heat maps depicting the correlation coefficient between DNA methylation (DNAm) probes in rows and metabolites in columns. Columns and rows were clustered using the complete linkage method. (a) Correlation between DNAm at pre-seroconversion (PSV) and metabolite measured at seroconversion (SV). (b) Correlation between DNAm at pre-seroconversion (PSV) and metabolite measured at seroconversion (SV).

\newpage

```{r results functions}
# Results functions
get_results = function(mod_list,q = 0.05,conf.m = "perc"){
  # Count number of significant parameters (at nominal 95% CI level)
  sig = lapply(names(mod_list),function(m){
    mod = mod_list[[m]]
    mod = tidy(mod,conf.int = T,conf.method = conf.m)
    if (mod[mod$term == "tnie","conf.low"] <= 0 & mod[mod$term == "tnie","conf.high"] >= 0){
      return(NA)
    } else {
      return(m)
    }
  })
  sig = sig[!is.na(sig)]
  k = length(sig)
  m = length(mod_list)
  BYci = 1-(k*q/m)
  # Recalculate CIs 
  r = lapply(sig, function(m){
    mod = mod_list[[m]]
    mod = tidy(mod,conf.int = T,conf.level = BYci,conf.method = conf.m)
    if (mod[mod$term == "tnie","conf.low"] <= 0 & mod[mod$term == "tnie","conf.high"] >= 0){
      return(NA)
    } else {
      return(c(strsplit(m," & ")[[1]],exp(mod[mod$term == "tnie","statistic"]),
               exp(mod[mod$term == "tnie","conf.low"]),
               exp(mod[mod$term == "tnie","conf.high"])))
    }
  })
  r = as.data.frame(rbindlist(r[!is.na(r)]))
  if (nrow(r)>0){
    colnames(r) = c("Probe","Metabolite","NIE","NIE CI Low","NIE CI High")
    r = r[order(r$NIE,decreasing = T),]
    rownames(r) = 1: nrow(r)
  } else {
    r = matrix(nrow = 0,ncol = 4)
    colnames(r) = c("Probe","Metabolite","NIE","NIE CI Low","NIE CI High")
  }
  r$Metabolite = metab_anno$name[match(r$Metabolite,metab_anno$feature)]
  r = r %>% select(Metabolite,everything())
  return(r)
}
# Annotation - requires output from get_results() and anno dataframe
annotate_results = function(r){
  methyl = r$Probe
  met = anno[anno$Name %in% methyl,c("chr","pos","Name","Relation_to_Island","UCSC_RefGene_Name")] %>%
    arrange(as.numeric(sub("chr","",chr)),pos)
  met = met %>% rename(Probe = "Name")
  return(met)
}
```

A total of 183 participants had both metabolite and methylation data at their PSV and SV study visits. Figure 1 outlines DAISY study visits. Participant characteristics are shown in Table 1, and there were no statistically significant differences between IA cases and controls with respect to age, sex, race/ethnicity, DR3/4 status, and first-degree relative (FDR) status.

## Table 1: Participant Characteristics at First Visit

```{r results='asis'}
demo = read.csv("./data/raw_data/demographics.csv")
# Pretty factor labels
demo$IAgroup2 = factor(demo$IAgroup2,levels = c("case","control"),labels = c("Case","Control"))
demo$NHW = factor(demo$NHW,levels = c(0,1),labels = c("No","Yes"))
demo$dr34 = factor(demo$dr34,levels = c(0,1),labels = c("No","Yes"))
demo$SEX = factor(demo$SEX)
demo$cohortgrp2 = factor(demo$cohortgrp2,levels = c("FDR","GP"),
                         labels = c("First-Degree Relative with T1D","General Population (no first degree relative with T1D)"))
# Table
t1 = tableby(IAgroup2 ~ anova(clinage,digits=1)+NHW+dr34+SEX+cohortgrp2,
             data = demo)
new_labels = list(clinage = "Age",NHW = "Non-Hispanic White",dr34 = "DR3/4",
                  SEX = "Sex",cohortgrp2 = "FDR Status")
summary(t1,labelTranslations = new_labels,pfootnote = T)
```

\newpage

Out of a total 379,557,915 possible pairs, our linear regression filtering step identified `r nrow(methyl_psv_candidates)` pairs with methylation measured at PSV and metabolite measured at SV where metabolite and methylation probe were significantly associated with one another and both associated with IA at the p < 0.01 level (Figure 1b). We also identified `r nrow(metab_psv_candidates)` pairs with metabolite measured at PSV and methylation measured at SV for mediation analysis (Figure 1c). Correlations between DNAm and metabolite for all candidates are shown in Figure 2.

After FCR adjustment, 7 pairs had a confidence interval for the estimate of NIE that did not contain 0 on the log odds scale (Table 2). Each pair contained a unique probe, but the metabolite histidine was the mediator in five of the pairs. No pairs in which metabolite was measured at PSV and methylation was measured at SV were significant after FCR adjustment. All confidence intervals are presented after FCR adjustment.

## Table 2: Methylation-Metabolite Pairs with Significant Natural Indirect Effects

```{r results='asis',warning=FALSE}
# Mediation results
metab = sapply(strsplit(names(methyl_psv_results)," & "),"[",2)
methyl_psv_no_unknowns = methyl_psv_results[-which(metab %in% unknowns)]
res = get_results(methyl_psv_no_unknowns)
# Annotation
met = annotate_results(res)
gene_cards = lapply(met$UCSC_RefGene_Name[which(met$UCSC_RefGene_Name != "")], function(g){
  genecard_description_summary[grep(g,genecard_description_summary$gene),]
})
gene_cards = do.call(rbind,gene_cards)
gene_cards$gene = as.character(gsub('\\"',"",gene_cards$gene))
gene_cards = gene_cards %>% select(gene,type,description,summary_genecard) %>%
  rename(UCSC_RefGene_Name = "gene",Type = "type",Description = "description",Summary = "summary_genecard")
gene_cards$UCSC_RefGene_Name = sub("-.*","",gene_cards$UCSC_RefGene_Name)
# Combine and format
res = left_join(res,met,by = "Probe")
res = full_join(res,gene_cards,by = "UCSC_RefGene_Name")
res = res %>% 
  select(Metabolite,Probe,chr,pos,Relation_to_Island,UCSC_RefGene_Name,
         Type,Description,NIE,`NIE CI Low`,`NIE CI High`) %>% 
  arrange(as.numeric(sub("chr","",chr))) %>% unite(Position,chr,pos,sep = ":") %>% 
  rename(Gene = "UCSC_RefGene_Name",`Relation to Island` = "Relation_to_Island")
# One row per pair
res = res %>% pivot_wider(values_from = c("Type","Description"),
                    values_fn = list(Type = ~paste(., collapse = "; "),
                                     Description = ~paste(., collapse = "; ")))
res[res=="NA"] = ""
# Print
kable(res,digits = 3)
```

\newpage

Six Reactome pathways were enriched (FDR q-value < 0.10) (Table 3), including plasma lipoprotein assembly, remodeling, and clearance and inositol phosphate metabolism. All pathways were enriched at least 2-fold.

## Table 3: Reactome Pathway Enrichment

```{r}
reactome = read.delim("./results/reactome_probe_background_top_100.txt")
reactome = reactome %>% arrange(final_probes_entrez_ids.csv..raw.P.value.) %>% 
  select(Reactome.pathways,background_entrez_ids.csv...REFLIST..16495.,
         final_probes_entrez_ids.csv..51.,final_probes_entrez_ids.csv..expected.,
         final_probes_entrez_ids.csv..fold.Enrichment.,final_probes_entrez_ids.csv..raw.P.value.)
colnames(reactome) = c("Term","# Genes in Reference List","# Genes in Top 100 Probes","Expected","Fold Enrichment","p value")
reactome = reactome %>% filter(`# Genes in Reference List` >= 5 & `# Genes in Top 100 Probes` >= 2)
reactome[,"FDR q value"] = p.adjust(reactome$`p value`,"fdr")
reactome = reactome[reactome$`FDR q value` < 0.1,]
# Import full reactome results
react_results <- fromJSON(file='./results/reactome_results.json')
# For each pathway, get gene IDs
genes = lapply(reactome$Pathway, function(g){
  i = agrep(sub(" \\(R-HSA-.*","",g),react_results$overrepresentation$result)[1]
  ids = react_results$overrepresentation$result[[i]]$input_list$mapped_id_list$mapped_id
  paste(as.character(lookUp(sub("GeneID:","",ids),'org.Hs.eg', 'SYMBOL')),collapse = ";")
})
reactome$Genes = unlist(genes)
# Order columns
reactome = reactome %>% select(Term,`# Genes in Top 100 Probes`,`# Genes in Reference List`,
                               Expected,`Fold Enrichment`,`p value`,`FDR q value`)
# Print
kable(reactome,digits = 3)
```

\newpage

None of the Biopathways were significant after p value adjustment (Table 4). However, several immune system-related pathways were significant at an unadjusted p < 0.05 level, including interleukin signaling and transmembrane transport of small molecules.

## Table 4: Bioplanet Enrichment (Top Ten)

```{r}
bioplanet = read.delim("./results/BioPlanet_2019_table.txt")
bioplanet = bioplanet %>% 
  separate(Overlap,c('Background','Ref'),sep = "/") %>%
  filter(Ref >= 2) %>%
  rename(`# Genes in Reference List` = Ref,`# Genes in Top 100 Probes` = Background,
         `p value`=P.value, `FDR q value`=Adjusted.P.value,
         `Fold Enrichment`=Odds.Ratio)
bioplanet$Old.P.value = NULL
bioplanet$Combined.Score = NULL
bioplanet$Old.Adjusted.P.value = NULL
kable(bioplanet[bioplanet$`p value` < 0.05,])
```

\newpage

# Discussion

We found seven unique pairs of CpG sites and metabolites with significant natural indirect mediation effects. These pairs included five DNAm probes mediated by histidine (a metabolite known to affect T1D risk), one probe (cg01604946) mediated by phostidyl choline p-32:0 or o-32:1, and one probe (cg00390143) mediated by sphingomyelin d34:2. For all seven pairs, the metabolite was positively associated with IA, and the methylation of the CpG site was negatively associated with both the metabolite and IA. Thus, an indirect effect less than one suggests that some of the protective effect of the CpG site is via reduction in the  metabolite levels. 

We have previously discovered nominally significant positive associations between histidine and the risk of progression from IA to T1D[@johnsonMetabolomicsrelatedNutrientPatterns2020], and other studies have indicated that the histidine-glutamate-glutamine pathway can be corrected by improving glycemic control.[@duttaImpactLongTermPoor2016] Also, histidine is a precursor to histamine, a monoamine that plays an important role in inflammation and has been linked to the development of T1D and other immune responses.[@alkanHistidineDecarboxylaseDeficiency2015;@mancaDifferentialMetabolomicSignatures2020] A knock-out mouse experiment for the gene encoding histidine decarboxylase (the enzyme that converts histidine to histamine) showed that decreased histamine levels were associated with a lower incidence of T1D and decreased levels of circulating IL-12 and IFN-$\gamma$.[@alkanHistidineDecarboxylaseDeficiency2015]

While the connection between histamine and T1D is relatively well-studied, it remains unclear how it might mediate the protective effects of the CpG sites we identified, and how the methylation of those CpG sites might affect T1D etiology. The association of IA and the CpG site profiled by probe cg19939773, located approximately 1.4 kb upstream of the second exon of the HLA-DQB2 gene, was mediated by histidine. HLA-DQB1 is an important risk gene in development of T1D, but the literature on HLA-DQB2 is less clear.[@husainMHCTypeDiabetes2008] Interestingly, methylation of the KIF26A gene (probe cg01172082), which was also mediated by histidine, was found to be affected by a 5-day high-fat high-energy diet in young men, and these changes could potentially contribute to insulin resistance seen in some of the study participants with T2D.[@gillbergAdiposeTissueTranscriptomics2016]

Das et al. found using whole exome sequencing (WES) that the *COL18A* gene (which we found was mediated by histidine) is protective against diabetic retinopathy, but the sample size was small and these results warrant replication.[@dasNovelGeneticVariants2018] A relatively large study also found that polymorphisms in the collagen protein encoded by COL18A may be related to increased obesity in patients with T2D.[@erreraCOL18A1HighlyExpressed2008] COL18A antisense RNA is also not well understood, but mouse models have provided some evidence that other antisense RNAs, namely GLUT-2, can increase risk of diabetes,[@valeraExpressionGLUT2Antisense1994] so it is possible that increased methylation results in lower expression of risk-increasing antisense RNA.

For PC (p-32:0) or PC (o-32:1) and SM (d34:2), a study by Oresic et al. found that participants who went on to develop T1D had lower levels of phosphatidyl cholines (PCs) at birth, but that "the lysoPC PC(18:0/0:0) was increased 1.5-fold within 9â€“18 mo before seroconversion."[@oresicDysregulationLipidAmino2008] Elevated lysoPCs are believed to be a marker of oxidative stress prior to the development of islet autoantibodies[@nokoffPathogenesisTypeDiabetes2013], but our group has also found that phosphatidylcholine (16:0_18:1(9Z)) was the strongest single metabolite predictor of IA reversion. The CpG site profiled by the cg01604946 probe is located within the SH3TC2 gene and its association with IA is mediated by the compound identified as PC (p-32:0) or PC (o-32:1). This phosphatidyl choline is generally associated with demyelinating motor and sensory neuropathy, and its connection to T1D requires more research.[@senderekMutationsGeneEncoding2003]

Like PCs, the research regarding sphingomyelins (SMs) generally suggests that they have a protective effect against developing T1D[@carryPhospholipidLevelsSeroconversion2021;@oresicDysregulationLipidAmino2008] and mouse models have shown that SM patches on pancreatic $\beta$ cells correlate well with insulin production capacity.[@kavishwarSphingomyelinPatchesPancreatic2013] However, SMs are also strongly associated with rapid eGFR decline[@onbehalfofthefinndianestudygroupSphingomyelinProgressionRenal2020] and general nephropathy[@makinenSphingomyelinAssociatedKidney2012] in those who already have T1D. Both SMs and PCs have also been linked to renal impairment and all-cause mortality in T1D,[@tofteLipidomicAnalysisReveals2019] which indicates that understanding of their role in T1D etiology remains incomplete and specific SMs and PCs may not be protective. Like the COL18A and KIF26A genes, which were mediated by histidine, the CpG probed by cg00390143 was associated with the GALNT9 gene which has been implicated in T2D but remains under-studied.[@liuKeyGenesAssociated2019]

This study has identified several potential causal pathways in the etiology of IA. The seven methylation sites that were significantly mediated by metabolites are potentially interesting candidates for elucidating epigenetic protection against T1D. While the causal pathways in our mediation models are temporally reasonable in the sense that the exposure was measured prior to the mediator, a better understanding of the biological pathways is necessary to confirm these exploratory analyses.

# Strengths and Limitations

Metabolites were measured in non-fasting samples, and these analyses did not account for dietary intake, which is the single biggest source of exposure to chemicals and nutrients.[@jonesNutritionalMetabolomicsProgress2012] However, we do not expect dietary intake to differ significantly between IA cases and controls. Enrichment and over-representation tools are not designed for the integration of methylation and metabolomics data. To our knowledge, there are no integrated enrichment or over-representation tools that are able to incorporate multiple omics datasets, so our biological interpretation relied on an ad-hoc choice of the 100 most significant DNAm probes.

Also, we did not adjust the candidate selection step for multiple testing, and only adjusted p values at the second stage, which could potentially lead to false positives. However, the best approach to p value adjustment for correlated variables in this filtering step remains an active area of statistical research and we suggest that this approach is unlikely to result in many false positives due to the adjustments used in the later stages.[@rudraPredictiveModelingMiRNAmediated2018]

However, despite these limitations we believe that our approach is a novel method for integration of omics data in epidemiological studies. The combination of a longitudinal study design and mediation analysis allows for causal interpretation of our results, which will hopefully guide additional research into biological mechanisms. Also, this approach is not limited to DNAm and metabolite studies, and could in theory be applied to any longitudinal study with multiple omics datasets.

\newpage

# Methods

## Study Design and Participants

Study participants were recruited via newborn screening at St. Joseph's Hospital in Denver, CO, USA and from unaffected first-degree relatives (FDR) of type 1 diabetes patients. For the DAISY nested case control study, IA cases were frequency-matched to controls by age at SV, race/ethnicity, and sample availability. The majority of participants were Non-Hispanic White (NHW), and race/ethnicity was categorized into NHW and Other for matching and analysis. All DAISY protocols were approved by the Colorado Multiple Institutional Review Board (COMIRB 92-080), and informed consent was obtained from all participants. All research was performed in accordance with relevant guidelines and regulations.[@johnsonLongitudinalDNAMethylation2020;@rewersBetaCellAutoantibodiesInfants1996;@rewersNewbornScreeningHLA1996] Participants with both methylation and metabolomic measures at the visit at PSV and SV were selected for these analyses (n=183).

## DNA Methylation

IA cases and frequency-matched controls were randomly assigned to either the 450 K group (which included duplicate samples for quality control) or EPIC group (which included replicates from the 450 K set for quality control). DNA methylation was profiled in peripheral whole blood using the Infinium HumanMethylation450K Beadchip (Illumina, San Diego, CA, USA, "450 K") for the 450 K set, and the Infinium HumanMethylation EPIC Beadchip ("EPIC") was used for the EPIC set. Both sets underwent identical pre-processing using the SeSAMe pipeline[@zhouSeSAMeReducingArtifactual2018], and measurement platform was included as a covariate in all statistical models in order to account for technological batch effects. Johnson and colleagues performed quality control and removed poor quality samples and DNAm probes (see Johnson et al.[@johnsonLongitudinalDNAMethylation2020] for details).  This resulted in 199,243 quality DNA methylation probes and  183 subjects having DNAm at both PSV and SV timepoints. Normalized M values were used in all statistical analyses.

## Metabolomics

Untargeted metabolomics were obtained using gas chromatography-time of flight mass spectrometry (GC-TOF MS), charged surface hybrid column quadrupole time of flight mass spectrometry (CSH-QTOF MS), and hydrophilic interaction chromatography quadrupole time of flight mass spectrometry (HILIC-QTOF MS), at the UC Davis West Coast Metabolomics Center. Non-fasting plasma samples were prepared and analyzed as previously described in Johnson et al. [@johnsonMetabolomicsrelatedNutrientPatterns2020]

For GC-TOF data, peak picking and annotation was performed using BinBase.[@skogersonVolatileCompoundBinBase2011] CSH-QTOF MS and HILIC-QTOF MS were processed using MS-Dial[@tsugawaMSDIALDataindependentMS2015], complex lipids were annotated with LipidBlast[@kindLipidBlastSilicoTandem2013] and Massbank of North America (http://mona.fiehnlab.ucdavis.edu/), and erroneous peaks were removed using MSFLO.[@defeliceMassSpectralFeature2017]

After collection, annotation, and post-processing, metabolomics data were normalized using SERRF[@fanSystematicErrorRemoval2019],a QC-based method designed to account for batch effects. Samples with low abundance (n = 2) and metabolites with a coefficient of variation more than two absolute deviations from the median (n = 344) were excluded, and data were transformed using the Box-Cox method.[@boxAnalysisTransformations1964] After processing and quality checks, 2457 untargeted metabolites remained, of which 1,905 metabolites were annotated with either a lab specific or InChIKey identifier.[@johnsonMetabolomicsrelatedNutrientPatterns2020] Only annotated metabolites were considered for these analyses.

## Statistical Analysis

All analyses were performed using the R programming environment[@rcoreteamLanguageEnvironmentStatistical2021] version 4.0.0 and included only participants with methylation and metabolomics data at both their SV and PSV study visits (n = 183). To reduce the number of methylation probe-metabolite pairs examined in the mediation analysis, we performed simple linear regression to find probes and metabolites that are correlated at a nominal p < 0.01 level. Next, we performed two independent logistic regressions on all significantly correlated pairs (one model for DNAm probe and one for metabolite) to determine pairs in which both were significantly associated with the IA outcome, again at a nominal 0.01 level. Pairs in which both metabolite and probe were correlated with one another, and both variables were associated with the outcome were selected for the mediation analysis. These p values were not adjusted for multiple comparisons because this was a candidate filtering step intended to save computing time for later steps. 

Model-based mediation analyses were performed using the regmedint package[@yoshidaRegmedintRegressionbasedCausal2020] version 0.2.1, an R implementation of Valeri and VanderWeele's SAS macros.[@valeriMediationAnalysisAllowing2013] We report the natural indirect effect (NIE). The natural indirect effect is interpreted on the odds scale, and represents the average change in outcome if the exposure $a$ (methylation at PSV) was fixed at 1, but the mediator $t$ (metabolite at SV) were changed from the level it would take if $a^*=0$ to the level it would take if $a=1$.[@vanderweeleExplanationCausalInference2015] In other words, it is the effect of the exposure on the outcome that operates through changing the mediator.[@vanderweeleExplanationCausalInference2015]

All regression models were adjusted for HLA-DR3/4 haplotype, age at PSV, time from PSV to SV, and sex, and included exposure mediator interaction terms as recommended by VanderWeele.[@vanderweeleExplanationCausalInference2015] Confidence intervals for the NIE were calculated based on 10,000 bootstrap simulations using the percentile method and adjusted using Benjamini and Yekutieli's False Coverageâ€“Statement Rate (FCR) method [@benjaminiFalseDiscoveryRate2005] at the q = 0.05 level (equivalent selection based on a false discovery rate-adjusted p value < 0.05). Because DAISY is a case-control study, cases are oversampled relative to the general population. To account for this, the regmedint package fits the mediator model using only the control participants, which approximates the results from a cohort study when the outcome is rare.[@vanderweeleExplanationCausalInference2015]

##  Biological Interpretation

Pathway over-representation was performed using PANTHER's[@miPANTHERVersion162021] implementation of Reactome pathways[@fabregatReactomePathwayKnowledgebase2018] and included 60 unique genes associated with the top 100 probes from the mediation step of the analysis. These p values were based on standard error of the NIE estimated using the multivariate delta method, as opposed to being converted from bootstrap confidence intervals.[@vanderweeleExplanationCausalInference2015] CpG sites were converted to Entrez IDs using the missMethyl R package[@phipsonMissMethylPackageAnalyzing2016] and the 199,243 original probes were used as the reference list. Pathways with less than 5 genes in the reference list and less than two genes in analysis list were excluded.

Pathway enrichment analysis was performed using the Enrichr (https://maayanlab.cloud/Enrichr/)[@xieGeneSetKnowledge2021] Biopathways 2019 module[@huangNCATSBioPlanetIntegrated2019], again with the 60 unique genes associated with the top 100 candidate probes. 

We also searched for known genotypes at certain loci with allele-specific methylation using the BIOS QTL browser (https://genenetwork.nl/biosqtlbrowser/)[@bonderDiseaseVariantsAlter2017]. These methylation quantitative trait loci (meQTLs) can alter methylation patterns across genomic regions.[@smithMethylationQuantitativeTrait2014]  We then used the biomaRt R package[@durinckMappingIdentifiersIntegration2009] to obtain gene symbols, which were annotated using GeneCards[@stelzerGeneCardsSuiteGene2016] via the GeneBook R package[@xuGeneBookMultitypesGene2019].

```{r eval=FALSE}
# Outcome and adjustment variables
ia = as.numeric(factor(psv$IAgroup2)) - 1
SEX = as.numeric(factor(psv$SEX)) - 1
dr34 = psv$dr34
age_delta = as.numeric(sv$clinage) - as.numeric(psv$clinage)
age = as.numeric(psv$clinage)
covariates = as.data.frame(cbind(ia,SEX,dr34,age,age_delta))
# Results
get_results = function(mod_list,q = 0.05,conf.m = "perc"){
  # Count number of significant parameters (at nominal 95% CI level)
  sig = lapply(names(mod_list),function(m){
    mod = mod_list[[m]]
    mod = tidy(mod,conf.int = T,conf.method = conf.m)
    if (mod[mod$term == "tnie","conf.low"] <= 0 & mod[mod$term == "tnie","conf.high"] >= 0){
      return(NA)
    } else {
      return(m)
    }
  })
  sig = sig[!is.na(sig)]
  k = length(sig)
  m = length(mod_list)
  BYci = 1-(k*q/m)
  # Recalculate CIs 
  r = lapply(sig, function(m){
    mod = mod_list[[m]]
    mod = tidy(mod,conf.int = T,conf.level = BYci,conf.method = conf.m)
    if (mod[mod$term == "tnie","conf.low"] <= 0 & mod[mod$term == "tnie","conf.high"] >= 0){
      return(NA)
    } else {
      return(c(strsplit(m," & ")[[1]],exp(mod[mod$term == "tnie","statistic"]),
               exp(mod[mod$term == "tnie","conf.low"]),
               exp(mod[mod$term == "tnie","conf.high"])))
    }
  })
  r = as.data.frame(rbindlist(r[!is.na(r)]))
  if (nrow(r)>0){
    colnames(r) = c("Probe","Metabolite","NIE","NIE CI Low","NIE CI High")
    r = r[order(r$NIE,decreasing = T),]
    rownames(r) = 1: nrow(r)
  } else {
    r = matrix(nrow = 0,ncol = 4)
    colnames(r) = c("Probe","Metabolite","NIE","NIE CI Low","NIE CI High")
  }
  r = r %>% select(Metabolite,everything())
  return(r)
}
res = get_results(methyl_psv_no_unknowns)
# Models
models = apply(res,1,function(r){
  methyl= as.character(r["Probe"])
  methyl = as.numeric(scale(psv[,methyl]))
  metab = as.character(r["Metabolite"])
  metab = as.numeric(scale(sv[,metab]))
  df = as.data.frame(cbind(methyl,metab,covariates))
  df = df[complete.cases(df),]
  d = df
  regmedint_obj = 
    regmedint(data = d,
              ## Variables
              yvar = "ia",
              avar = "methyl",
              mvar = "metab",
              cvar = c("SEX","dr34","age","age_delta"),
              ## Values at which effects are evaluated
              a0 = 0,
              a1 = 1,
              m_cde = 1,
              c_cond = c(1,1,1,1),
              ## Model types
              mreg = "linear",
              yreg = "logistic",
              ## Additional specification
              interaction = T,
              casecontrol = T)
})
med_package = apply(res,1,function(r){
  methyl= as.character(r["Probe"])
  methyl = as.numeric(scale(psv[,methyl]))
  metab = as.character(r["Metabolite"])
  metab = as.numeric(scale(sv[,metab]))
  df = as.data.frame(cbind(methyl,metab,covariates))
  df = df[complete.cases(df),]
  ymod = glm(formula = ia ~ methyl * metab + SEX + dr34 + age + age_delta, 
    family = binomial(link = "logit"), data = df)
  mmod = lm(formula = metab ~ methyl + SEX + dr34 + age + age_delta, data = df)
  med = mediate(mmod,ymod,10000,treat = "methyl",mediator = "metab")
})
```

\newpage

# References
