---
title: "A Mediation Approach to Discovering Causal Relationships Between the Metabolome and DNA Methylation in Type 1 Diabetes"
date: ""
author:
- Timothy Vigers, B.A.^[UCD], Lauren A. Vanderlinden, M.S., Randi K. Johnson, Ph.D., Patrick M. Carry, M.S., Alexander M. Kaizer, Ph.D., Jill M. Norris, Ph.D., Katerina Kechris, Ph.D.
bibliography: "/Users/timvigers/Dropbox/Miscellaneous/references.bib"
csl: "/Users/timvigers/Dropbox/Miscellaneous/american-medical-association.csl"
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = "~/Dropbox/School/MS Thesis")
library(arsenal)
```

# Abstract

## Objective:

## Methods:

## Results:

## Conclusion:

\newpage

# Introduction

Type 1 diabetes (T1D) is an autoimmune disease characterized by the production of antibodies which target pancreatic $\beta$-cells. This leads to both micro and macrovascular complications, and individuals with T1D are at an increased risk of experiencing a major cardiac event (e.g. stroke, angina, and myocardial infarction) and are less likely to have satisfactory outcomes after a coronary event.[@atkinsonTypeDiabetes2014] The disease currently affects over 30 million people worldwide,[@rakyanIdentificationTypeDiabetes2011] and is increasing by 3-4% per year on average.[@tuomilehtoEmergingGlobalEpidemic2013] However, the global burden of disease is difficult to estimate due to geographic variation in incidence.[@atkinsonTypeDiabetes2014;@rakyanIdentificationTypeDiabetes2011]

Genetic predisposition accounts for some of the etiology of T1D (sibling relative risk has been estimated at 15)[@polychronakosUnderstandingTypeDiabetes2011] and explains some geographic variation in incidence.[@tuomilehtoEmergingGlobalEpidemic2013] Human leukocyte antigen (HLA) genes were the first to be linked to T1D and account for much of the genetic predisposition to the disease, but genome-wide association studies (GWAS) have also identified more than 40 other loci associated with increased relative risk of T1D. However, it is still unclear how these multiple loci interact with one another and the environment to produce a T1D diagnosis. In addition to the complex genetics of T1D, low monozygotic (MZ) twin concordance (approximately 50%) and studies of migrant populations support the theory that non-genetic factors play an important role in T1D development.[@rakyanIdentificationTypeDiabetes2011] 

Epigenetic differences may be important contributors to non-genetic T1D etiology. Changes in methylation have been associated with other autoimmune conditions, and monozygotic twins can be epigenetically heterogeneous despite sharing identical genetic code.[@rakyanIdentificationTypeDiabetes2011]  Rakyan et al. and Stefan et al. performed epigenome-wide association studies (EWAS) in discordant and concordant twin pairs and found that methylation profiles were more similar among participants with T1D than to unaffected twins. Epigenetics profiles were also combined with GWAS data and differentially expressed methylation sites were mapped to 6 well known T1D susceptibility genes, including two major histocompatibility complex (MHC) genes and several HLA loci.[@stefanDNAMethylationProfiles2014;@rakyanIdentificationTypeDiabetes2011] Finally, the Norris group has found that T1D cases had different longitudinal methylation patterns compared to controls prior to diagnosis.[@johnsonLongitudinalDNAMethylation2020] 

Environmental factors including viruses, diet, and the metabolome have also been linked with islet autoimmunity[@oresicDysregulationLipidAmino2008] and T1D etiology.[@atkinsonTypeDiabetes2014] Previous studies have found associations between T1D and differentially expressed phospholipids and sphingolipids, excretion of modified amino acids, and vitamin D (and related compounds on its metabolic pathway).[@frohnertMetabolomicsChildhoodDiabetes2016;@atkinsonTypeDiabetes2014] Branched chain amino acids (BCAAs) have also been associated with insulin resistance.[@guasch-ferreMetabolomicsPrediabetesDiabetes2016]

Although the effects of the metabolome and DNA methylation have been studied separately in T1D, the combined nutri-epigenetics of T1D remain unclear. Early in-vitro studies confirmed metabolome-dependent alterations in DNA methylation associated with various cancers,[@chiacchieraEpigeneticMethylationsTheir2013;@putluriMetabolomicProfilingReveals2011] and even identified oncometabolites associated with the development of glioma.[@wellerIsocitrateDehydrogenaseMutations2011] More recent human studies also support the connection between the metabolome and methylation in breast cancer,[@alakwaaBioinformaticsAnalysisMetabolomics2020] colorectal cancer,[@wangMultiomicProfilingReveals2020] and smoking-related diseases.[@huangUntargetedMetabolomicsReveals2018] A 2018 study by Zaghlool et al. used linear models to link metabolomics and DNA methylation with type 2 diabetes (T2D) and obesity, and Mendelian randomization (MR) analyses suggest "a causal effect of metabolite levels on methylation of obesity-associated CpG sites."[@zaghloolDeepMolecularPhenotypes2018] 

Integration of epigenetic and metabolomics data requires statistical methods capable of evaluating associations between many different variables in complex multilevel networks. Bayesian networks are becoming increasingly popular for analysis of large-scale systems biology ("omics") data, including protein-protein interactions and gene regulatory networks.32 This approach can be used to generate intuitive graphical models, which represent probabilistic dependence between multiple variables32,33 and avoid many of the pitfalls of traditional mediation analyses.34 Also, probabilistic dependence relationships can be interpreted as causal pathways, which clarifies the biological interpretation of results. We hope to derive a graphical model of the relationships between metabolites and methylation sites, to illustrate and make inferences on the epigenetic and environmental pathways that lead to T1D.

\newpage

# Methods

## Study Design and Participants

The Diabetes Autoimmunity Study in the Young (DAISY) cohort follows 2547 high-risk children in Colorado for the development of IA and T1D. Study participants were recruited via newborn screening at St. Joseph's Hospital in Denver, CO, USA and from unaffected first-degree relatives (FDR) of type 1 diabetes patients. The study follows participants prospectively and includes blood sample collection at 9, 15, and 24 months, then annual collection until islet autoimmunity (IA) is detected. IA is defined as the second consecutive visit at which a confirmed auto-antibody to insulin, GAD65, IA-2, or ZnT8 was detected.2 Participants who develop IA are asked to follow an accelerated protocol with visits and blood sample collection every 3-6 months, until they are diagnosed with diabetes by a physician. T1D cases were matched to controls by age at SV, race/ethnicity, and sample availability. The majority of participants were Non-Hispanic White (NHW), and race/ethnicity was categorized into NHW and Other for matching and analysis. All DAISY protocols were approved by the Colorado Multiple Institutional Review Board (COMIRB 92-080), and informed consent was obtained from all participants. All research was performed in accordance with relevant guidelines and regulations.[@johnsonLongitudinalDNAMethylation2020]

Participants with both methylation and metabolomic measures at the visit at which seroconversion was detected (SV) and the visit immediately prior (pre-seroconversion or PSV) were selected for these analyses. 

## DNA Methylation

T1D cases and matched controls were randomly assigned to either the 450 K group (which included duplicate samples for quality control) or EPIC group (which included replicates from the 450 K set for quality control). DNA methylation was profiled in peripheral whole blood using the Infinium HumanMethylation450K Beadchip (Illumina, San Diego, CA, USA, "450 K") for the 450 K set, and the Infinium HumanMethylation EPIC Beadchip ("EPIC") was used for the EPIC set. Both sets underwent identical pre-processing using the SeSAMe pipeline[@zhouSeSAMeReducingArtifactual2018], and measurement platform was included as a covariate in all statistical models in order to account for technological batch effects.

The SeSAMe pipeline was implemented using the sesame package (v1.0.0) in R (v3.5.2), and consists of Noob normalization[@tricheLowlevelProcessingIllumina2013] , non-linear dye bias correction, and pOOBAH to help identify failed hybridization. Next, arrays with high proportion of failed probes (100,000 failed probes for 450 K and 200,000 failed probes for EPIC) were removed (450 K = 6, EPIC = 2), along with arrays with discordant predicted sex and clinically recorded sex (450 K = 5, EPIC = 2) Probes missing in at least 1 sample were also removed from subsequent analyses (450 K = 109,956, EPIC = 202,222). ComBat[@leekSvaPackageRemoving2012] was used to remove batch effects using plate and row to delineate batch.

## Metabolomics

Untargeted metabolomics were obtained using gas chromatography-time of flight mass spectrometry (GC-TOF MS), charged surface hybrid column quadrupole time of flight mass spectrometry (CSH-QTOF MS), and hydrophilic interaction chromatography quadrupole time of flight mass spectrometry (HILIC-QTOF MS), at the NIH West Coast Metabolomics Center at the University of California, Davis. Non-fasting plasma samples were prepared and analyzed as previously described in Johnson et al. [@johnsonMetabolomicsrelatedNutrientPatterns2020]

For GC-TOF data, peak picking and annotation was performed using BinBase.[@skogersonVolatileCompoundBinBase2011] CSH-QTOF MS and HILIC-QTOF MS were processed using MS-Dial[@tsugawaMSDIALDataindependentMS2015] and complex lipids were annotated with LipidBlast[@kindLipidBlastSilicoTandem2013] and Massbank of North America (http://mona.fiehnlab.ucdavis.edu/), and erroneous peaks were removed using MSFLO.[@defeliceMassSpectralFeature2017]

After collection, annotation, and postprocessing, metabolomics data were normalized using SERRF[@fanSystematicErrorRemoval2019],a QC-based method designed to account for batch effects. Samples with low abundance (n = 2) and metabolites with a coefficient of variation more than two absolute deviations from the median (n = 344) were excluded, and data were transformed using the box-cox method.[@boxAnalysisTransformations1964] After processing and quality checks, 2457 untargeted metabolites (including 1905 annotated) remained.[@johnsonMetabolomicsrelatedNutrientPatterns2020]

## Statistical Analysis

All analyses were performed using the R programming environment[@rcoreteamLanguageEnvironmentStatistical2021] and included only participants with methylation and metabolomics data at both their SV and PSV study visits (n = 183). Because regression-based mediation analyses test hypotheses about causal networks, the approach is only valid if the causal assumptions are appropriate.[@fiskeHandbookSocialPsychology2010] As such, we only considered mediation models in which the methylation probe and metabolite were temporally separated. For example, it is more reasonable to assume that a methylation site measured at PSV has a causal relationship with a metabolite measured at the SV visit (or vice versa) than to assume a causal relationship between two variables measured at the same time.

To reduce the number of methylation probe-metabolite pairs examined in the mediation analysis, we performed simple linear regression to find probes and metabolites that are correlated at a nominal p < 0.01 level. Next, we performed independent logistic regression on all significantly correlated pairs to determine those that were significantly associated with the IA outcome, again at a nominal 0.01 level. Pairs in which both metabolite and probe were correlated with one another, and both variables were associated with the outcome were selected for the mediation analysis. These p values were not adjusted for multiple comparisons because this was simply a candidate filtering step intended to save computing time later on. We selected 1490 pairs with methylation measured at PSV and metabolite measured at SV and 600 pairs with metabolite measured at PSV and methylation measured at SV for further investigation.

Model-based mediation analyses were performed using the regmedint package[@yoshidaRegmedintRegressionbasedCausal2020], an R implementation of Valeri and VanderWeele's SAS macros.[@valeriMediationAnalysisAllowing2013] We report the natural indirect effect (NIE), which can be interpreted as the effect of the exposure on the outcome that operates through changing the mediator.[@vanderweeleExplanationCausalInference2015] All regression models were adjusted for HLA-DR3/4 haplotype, age, time from PSV to SV, and sex. Confidence intervals for the NIE were calculated based on 10,000 bootstrap simulations using the percentile method.

# Results

## Table 1: Participant Characteristics at PSV

```{r}
demo = read.csv("./data/raw_data/demographics.csv")
# Pretty factor labels
demo$IAgroup2 = factor(demo$IAgroup2,levels = c("case","control"),labels = c("Case","Control"))
demo$NHW = factor(demo$NHW,levels = c(0,1),labels = c("No","Yes"))
demo$dr34 = factor(demo$dr34,levels = c(0,1),labels = c("No","Yes"))
demo$SEX = factor(demo$SEX)
demo$cohortgrp2 = factor(demo$cohortgrp2,levels = c("FDR","GP"),
                         labels = c("First-Degree Relative","General Population"))
# Table
t1 = tableby(IAgroup2 ~ clinage+NHW+dr34+SEX+cohortgrp2,
             data = demo)
new_labels = list(clinage = "Age",NHW = "Non-Hispanic White",dr34 = "DR3/4",
                  SEX = "Sex",cohortgrp2 = "FDR Status")
summary(t1,labelTranslations = new_labels)
```

# Discussion

- Limitations:
  - Non-fasting plasma samples for metabolomics, and these analyses did not account for dietary patterns.

\newpage

# References
