---
title: "MS Thesis: Mediation"
author: "Tim Vigers"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,message = FALSE)
knitr::opts_knit$set(root.dir = "~/Dropbox/School/MS Thesis")
options(connectionObserver = NULL)
library(biomaRt)
library(missMethyl)
library(GeneBook)
library(data.table)
library(patchwork)
library(DiagrammeR)
library(knitr)
library(tidyverse)
library(regmedint)
library(broom)
```


```{r}
load("./data/raw_data/psv_sv_dataset.Rdata")
load("./data/mediation/methyl_psv_candidates_p_01.Rdata")
load("./data/mediation/metab_psv_candidates_p_01.Rdata")
load("./data/mediation/methyl_psv_both_scaled_results.Rdata")
load("./data/mediation/metab_psv_both_scaled_results.Rdata")
load("./data/raw_data/annotation.850K.Rdata")
anno = as.data.frame(anno)
# Import and format metabolite annotation
gctof_anno = read.csv("./data/raw_data/gctof.featureAnno.csv") %>% 
  select(feature_name,BinBase.name) %>%
  rename(feature = feature_name,name = BinBase.name)
hilic_anno = read.csv("./data/raw_data/hilic.featureAnno.csv") %>%
  select(feature_name,Metabolite.name) %>%
  rename(feature = feature_name,name = Metabolite.name)
lipid_anno = read.csv("./data/raw_data/lipid.featureAnno.csv") %>%
  select(feature_name,Metabolite.name) %>%
  rename(feature = feature_name,name = Metabolite.name)
oxy_anno = read.csv("./data/raw_data/oxylipin.featureAnno.csv") %>%
  select(cleanedDat_name,full.name) %>%
  rename(feature = cleanedDat_name,name = full.name)
vitd_anno = read.csv("./data/raw_data/vitD.featureAnno.csv") %>% 
  select(featureName,originalName) %>%
  rename(feature = featureName,name = originalName)
# Combine
metab_anno = bind_rows(gctof_anno,hilic_anno,lipid_anno,oxy_anno,vitd_anno)
```

# Pair Selection

To look for candidate metabolite-methylation pairs, I ran three linear models and selected based on the p value cutoffs below (e.g. metabolite and methylation associated at p < 0.01, and both associated with IA status at p < 0.01).

```{r fig.height=8,fig.width=8}
grViz("
  graph{
  node [shape = box,fontname = Helvetica]
  a[label= 'Metabolite at SV']
  b[label= 'Methylation at PSV']
  c[label= 'IA']
  a -- b [label = 'p < 0.01',fontsize = 9]
  a -- c [label = 'p< 0.01',fontsize = 9]
  {rank = same;b -- c [label = 'p< 0.01',fontsize = 9]}
  }
")
```

Using this mediation structure, there were `r nrow(methyl_psv_candidates)` pairs selected with `r length(unique(methyl_psv_candidates[,2]))` unique metabolites.

```{r fig.height=8,fig.width=8}
grViz("
  graph{
  node [shape = box,fontname = Helvetica]
  a[label= 'Methylation at SV']
  b[label= 'Metabolite at PSV']
  c[label= 'IA']
  a -- b [label = 'p < 0.01',fontsize = 9]
  a -- c [label = 'p< 0.01',fontsize = 9]
  {rank = same;b -- c [label = 'p< 0.01',fontsize = 9]}
  }
")
```

Using this mediation structure, there were `r nrow(metab_psv_candidates)` pairs selected with `r length(unique(metab_psv_candidates[,2]))` unique metabolites.

# Mediation Analysis

## Methylation at PSV

### Models

Terms:

$$
Y = \text{IA Status}\\
M = \text{Metabolite at SV}\\
T = \text{Methylation at PSV}\\
X_2 = \text{Sex}\\
X_3 = \text{DR3/4 Status}\\
X_4 = \text{Age at PSV}\\
X_5 = \text{Time from PSV to SV}\\
\text{i indexes subject}\\
$$

#### Mediator Model

$$
M_i = \beta_0 + \beta_1 T_i+\beta_2'C+\epsilon_i\\
\text{With } \epsilon_i\text{ i.i.d. } N(0,\sigma^2)
$$

In order to account for the case control design, the mediator model is fit to only the control subjects. The nature of case controls studies means that the cases are over-sampled compared to controls. The regression parameters in the outcome model are consistently estimated in logistic regression, but the study design needs to be taken into account in the mediator model. Fitting this model using only the controls is a good approximation for the regression coefficients that would be obtained in a cohort study, assuming the outcome is rare and that inclusion in the study depends only on the outcome (and not the mediator or exposure).

#### Outcome Model

$$
\text{Let }p=\text{the probability a given subject is an IA case}=P(Y = 1)\\
ln(\frac{p}{1-p}) = \theta_0 +\theta_1 T_i+\theta_2 M_i+\theta_3(T_i*M_i)+\theta_4'C
$$

Where $\beta_2'$ and $\theta_4'$ are vectors of estimates for multiple confounders $C$.

### Mediation

CDE is the controlled direct effect. This is how much the outcome would change on average if the mediator was fixed (at m) across the whole population, and the "treatment" (exposure) changed from 0 to 1. 

NDE is the natural direct effect. It represents the average change in outcome if the "exposure" changed from 0 to 1, and each individual's value for the mediator varied as it normally would (in the absence of the exposure). In other words, it is the effect of the exposure on the outcome that would remain if the path from exposure to mediator was removed. 

NIE is the natural indirect effect. It is the average change in outcome if the exposure was fixed at 1, but "the mediator were changed from the level it would take if $t^*=0$ to the level it would take if $t=1$." This is essentially the effect of the exposure on the outcome that operates through changing the mediator. 

Note, if the interaction term is 0, both the CDE and NDE are equal to the direct effect under Baron and Kenny's formulation. Also, while CDE, NDE, and NIE are interpreted on the odds scale, proportion mediated is interpreted on the risk scale.

$$
OR^{CDE}(m)=exp(\theta_1+\theta_3 m)\\
OR^{NDE}=exp((\theta_1+\theta_3\beta_0+\theta_3\beta_1+\theta_3\beta_2'C+\theta_3\theta_2\sigma^2)+0.5\theta_3^2\sigma^2)\\
OR^{NIE}=exp(\theta_2\beta_1+\theta_3\beta_1)\\
\text{Proportion mediated (PM)}=\frac{OR^{NDE}(OR^{NIE}-1)}{(OR^{NDE}*OR^{NIE}-1)}
$$

## Results

In the `regmedint` summary table, "pure natural direct effect (pnde) is what is typically called the natural direct effect (NDE). The total natural indirect effect (tnie) is the corresponding natural indirect effect (NIE)." [link](https://kaz-yos.github.io/regmedint/)

Because proportion mediated is a highly variables measure and is problematic when the NDE and NIE are in opposite directions, it is generally best to use the estimate and confidence interval for the NIE instead (pg. 48).

```{r include=FALSE}
# Results functions
get_results = function(mod_list,q = 0.05,conf.m = "perc"){
  # Count number of significant parameters (at nominal 95% CI level)
  sig = lapply(names(mod_list),function(m){
    mod = mod_list[[m]]
    mod = tidy(mod,conf.int = T,conf.method = conf.m)
    if (mod[mod$term == "tnie","conf.low"] <= 0 & mod[mod$term == "tnie","conf.high"] >= 0){
      return(NA)
    } else {
      return(m)
    }
  })
  sig = sig[!is.na(sig)]
  k = length(sig)
  m = length(mod_list)
  BYci = 1-(k*q/m)
  # Recalculate CIs 
  r = lapply(sig, function(m){
    mod = mod_list[[m]]
    mod = tidy(mod,conf.int = T,conf.level = BYci,conf.method = conf.m)
    if (mod[mod$term == "tnie","conf.low"] <= 0 & mod[mod$term == "tnie","conf.high"] >= 0){
      return(NA)
    } else {
      return(c(m,exp(mod[mod$term == "tnie","statistic"]),
               exp(mod[mod$term == "tnie","conf.low"]),
               exp(mod[mod$term == "tnie","conf.high"])))
    }
  })
  r = as.data.frame(rbindlist(r[!is.na(r)]))
  if (nrow(r)>0){
    colnames(r) = c("Pair","NIE","NIE CI Low","NIE CI High")
    r = r[order(r$NIE,decreasing = T),]
    rownames(r) = 1: nrow(r)
    return(r)
  } else {
    r = matrix(nrow = 0,ncol = 4)
    colnames(r) = c("Pair","NIE","NIE CI Low","NIE CI High")
    return(r)
  }
}
# Annotation - requires output from get_results() and anno dataframe
annotate_results = function(r){
  metab = sapply(strsplit(r$Pair," & "),"[",2)
  print(kable(metab_anno[metab_anno$feature %in% metab,]))
  methyl = sapply(strsplit(r$Pair," & "),"[",1)
  met = anno[anno$Name %in% methyl,c("chr","pos","Name","Probe_rs","Relation_to_Island","UCSC_RefGene_Name")] %>%
    arrange(as.numeric(sub("chr","",chr)),pos)
  return(met)
}
```

### Methylation at PSV

```{r results='asis'}
res = get_results(methyl_psv_results)
kable(res,digits = 3)
```

#### Annotation

```{r results='asis'}
met = annotate_results(res)
kable(met)
```

```{r}
gene_cards = lapply(met$UCSC_RefGene_Name[which(met$UCSC_RefGene_Name != "")], function(g){
  genecard_description_summary[grep(g,genecard_description_summary$gene),]
})
gene_cards = do.call(rbind,gene_cards)
kable(gene_cards,row.names = F)
```

#### SNPs

rs IDs are from the BIOS QTL browser (`genenetwork.nl/biosqtlbrowser/`)

```{r message=FALSE,cache=TRUE}
# Import BIOS QTL information
methyl = sapply(strsplit(res$Pair," & "),"[",1)
snps = read.delim("~/Dropbox/Work/Methylation/BIOS QTL/2015_09_02_cis_meQTLsFDR0.05-CpGLevel.txt")
snps = snps[snps$ProbeName %in% methyl,]
rsids = snps$SNPName

ensembl_snp = useMart("ENSEMBL_MART_SNP", dataset="hsapiens_snp")
bm = getBM(attributes = c('refsnp_id',"refsnp_source",'chr_name','chrom_start',
                          'chrom_end',"associated_gene","phenotype_name","phenotype_description",
                          "consequence_type_tv",
                          "clinical_significance"),
           filters = 'snp_filter', values = rsids,mart = ensembl_snp)
bm$Probe = snps$ProbeName[match(bm$refsnp_id,snps$SNPName)]

bm = bm %>% mutate(chr_num = suppressWarnings(as.numeric(chr_name))) %>% 
  arrange(chr_num,chrom_start) %>% select(Probe,everything())
kable(bm,row.names = F)
```

#### Panther

I used the `missMethyl` R package to find Entrez IDs for the significant probes from above. These IDs were then used for gene list overrepresentation analysis. For the background list of all CpG sites tested, I used the list of probes that were selected for mediation testing during the pair selection step.

```{r cache=TRUE}
# Write ensembl IDs for significant and background probes
ids = getMappedEntrezIDs(methyl)$sig.eg
background = unique(methyl_psv_candidates[,1])
background = getMappedEntrezIDs(background)$sig.eg
write.table(paste0("GeneID:",ids),
            file = "~/Dropbox/School/MS Thesis/results/final_probes_entrez_ids.csv",
            row.names = F,col.names = F,sep = ",",quote = F)
write.table(paste0("GeneID:",background),
            file = "~/Dropbox/School/MS Thesis/results/background_entrez_ids.csv",
            row.names = F,col.names = F,sep = ",",quote = F)
```

```{r}
reactome = read.delim("./results/reactome_overrepresentation.txt")
reactome$final_probes_entrez_ids.csv..raw.P.value. = 
  as.numeric(sub("<","",reactome$final_probes_entrez_ids.csv..raw.P.value.))
```

There were no significant results using the Panther pathways, even without adjustment for multiple tests. Using the Reactome pathways, there were `r sum(reactome$final_probes_entrez_ids.csv..raw.P.value. <= 0.05)` pathways that were significantly over-represented (based on the binomial test). However, none of these were significant after FDR adjustment for multiple comparisons.

```{r print reactome}
kable(reactome[reactome$final_probes_entrez_ids.csv..raw.P.value. <= 0.05,])
```

#### Enrichr

```{r enrichr,cache=TRUE}
library(org.Hs.eg.db)
genes = mapIds(org.Hs.eg.db, ids, 'SYMBOL', 'ENTREZID')
write.table(genes,file = "~/Dropbox/School/MS Thesis/results/final_entrez_symbols.csv",
            row.names = F,col.names = F,sep = ",",quote = F)
```

Again, the 18 significant probes were used for Enrichr analysis, the results of which can be found here: <https://maayanlab.cloud/Enrichr/enrich?dataset=e23adb5dffdee89b1d30b13107c3bfb7>. The Bioplanet 2019 results here look particularly interesting, but it's tough to say how much to read into them.

#### HMDB

```{r hmdb}
metab = sapply(strsplit(res$Pair," & "),"[",2)
names = metab_anno$name[match(metab,metab_anno$feature)]

hmdb = read.delim("~/Dropbox/Work/Metabolomics/HMDB/HMDB_table.txt")

```

### Metabolite at PSV

Before adjusting for multiple comparisons, there were 13 significant pairs. However, none were significant after 95% false coverage rate (FCR) adjustment.