---
title: "MS Thesis: Mediation"
author: "Tim Vigers"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,message = FALSE)
knitr::opts_knit$set(root.dir = "/Users/timvigers/Dropbox/School/MS Thesis")
library(patchwork)
library(DiagrammeR)
library(knitr)
library(tidyverse)
library(regmedint)
library(blandr)
```


```{r}
load("./data/mediation/methyl_psv_candidates_p_01.Rdata")
load("./data/mediation/metab_psv_candidates_p_01.Rdata")
load("./data/mediation/methyl_psv_results.Rdata")
load("./data/mediation/metab_psv_results.Rdata")
load("./data/raw_data/annotation.850K.Rdata")
anno = as.data.frame(anno)
lipid_anno = read.csv("./data/raw_data/lipid.featureAnno.csv")
```

# Pair Selection

To look for candidate metabolite-methylation pairs, I ran three linear models and selected based on the p value cutoffs below (e.g. metabolite and methylation associated at p < 0.01, and both associated with IA status at p < 0.01).

```{r fig.height=8,fig.width=8}
grViz("
  graph{
  node [shape = box,fontname = Helvetica]
  a[label= 'Metabolite at SV']
  b[label= 'Methylation at PSV']
  c[label= 'IA']
  a -- b [label = 'p < 0.01',fontsize = 9]
  a -- c [label = 'p< 0.01',fontsize = 9]
  {rank = same;b -- c [label = 'p< 0.01',fontsize = 9]}
  }
")
```

Using this mediation structure, there were `r nrow(methyl_psv_candidates)` pairs selected with `r length(unique(methyl_psv_candidates[,2]))` unique metabolites.

```{r}
kable(lipid_anno[lipid_anno$feature_name %in% unique(methyl_psv_candidates[,2]),])
```

```{r fig.height=8,fig.width=8}
grViz("
  graph{
  node [shape = box,fontname = Helvetica]
  a[label= 'Methylation at SV']
  b[label= 'Metabolite at PSV']
  c[label= 'IA']
  a -- b [label = 'p < 0.01',fontsize = 9]
  a -- c [label = 'p< 0.01',fontsize = 9]
  {rank = same;b -- c [label = 'p< 0.01',fontsize = 9]}
  }
")
```

Using this mediation structure, there were `r nrow(metab_psv_candidates)` pairs selected with `r length(unique(metab_psv_candidates[,2]))` unique metabolites.

```{r}
kable(lipid_anno[lipid_anno$feature_name %in% unique(metab_psv_candidates[,2]),])
```

Some of the hilic metabolites are lacking annotation information.

# Mediation Analysis

## Methylation at PSV

### Models

Terms:

$$
Y = \text{IA Status}\\
M = \text{Metabolite at SV}\\
T = \text{Methylation at PSV}\\
X_2 = \text{Sex}\\
X_3 = \text{DR3/4 Status}\\
X_4 = \text{Age at PSV}\\
X_5 = \text{Time from PSV to SV}\\
\text{i indexes subject}\\
$$

#### Mediator Model

$$
M_i = \beta_0 + \beta_1 T_i+\beta_2'C+\epsilon_i\\
\text{With } \epsilon_i\text{ i.i.d. } N(0,\sigma^2)
$$

In order to account for the case control design, the mediator model is fit to only the control subjects. The nature of case controls studies means that the cases are over-sampled compared to controls. The regression parameters in the outcome model are consistently estimated in logistic regression, but the study design needs to be taken into account in the mediator model. Fitting this model using only the controls is a good approximation for the regression coefficients that would be obtained in a cohort study, assuming the outcome is rare and that inclusion in the study depends only on the outcome (and not the mediator or exposure).

#### Outcome Model

$$
\text{Let }p=\text{the probability a given subject is an IA case}=P(Y = 1)\\
ln(\frac{p}{1-p}) = \theta_0 +\theta_1 T_i+\theta_2 M_i+\theta_3(T_i*M_i)+\theta_4'C
$$

Where $\beta_2'$ and $\theta_4'$ are vectors of estimates for multiple confounders $C$.

### Mediation

CDE is the controlled direct effect. This is how much the outcome would change on average if the mediator was fixed (at m) across the whole population, and the "treatment" (exposure) changed from 0 to 1. 

NDE is the natural direct effect. It represents the average change in outcome if the "exposure" changed from 0 to 1, and each individual's value for the mediator varied as it normally would (in the absence of the exposure). In other words, it is the effect of the exposure on the outcome that would remain if the path from exposure to mediator was removed. 

NIE is the natural indirect effect. It is the average change in outcome if the exposure was fixed at 1, but "the mediator were changed from the level it would take if $t^*=0$ to the level it would take if $t=1$." This is essentially the effect of the exposure on the outcome that operates through changing the mediator. 

Note, if the interaction term is 0, both the CDE and NDE are equal to the direct effect under Baron and Kenny's formulation. Also, while CDE, NDE, and NIE are interpreted on the odds scale, proportion mediated is interpreted on the risk scale.

$$
OR^{CDE}(m)=exp(\theta_1+\theta_3 m)\\
OR^{NDE}=exp((\theta_1+\theta_3\beta_0+\theta_3\beta_1+\theta_3\beta_2'C+\theta_3\theta_2\sigma^2)+0.5\theta_3^2\sigma^2)\\
OR^{NIE}=exp(\theta_2\beta_1+\theta_3\beta_1)\\
\text{Proportion mediated (PM)}=\frac{OR^{NDE}(OR^{NIE}-1)}{(OR^{NDE}*OR^{NIE}-1)}
$$

## Results

In the `regmedint` summary table, "pure natural direct effect (pnde) is what is typically called the natural direct effect (NDE). The total natural indirect effect (tnie) is the corresponding natural indirect effect (NIE)." [link](https://kaz-yos.github.io/regmedint/)

Because proportion mediated is a highly variables measure and is problematic when the NDE and NIE are in opposite directions, it is generally best to use the estimate and confidence interval for the NIE instead (pg. 48).

### Methylation at PSV

```{r}
r = lapply(names(methyl_psv_results),function(m){
  mod = methyl_psv_results[[m]]
  s = summary(mod,exponentiate = T)$summary_myreg
  if (s["tnie","p"] < 0.05){return(m)}else{return(NA)}
})
r = unlist(r[!is.na(r)])
```

There are `r length(r)` pairs with significant NIE:

```{r results='asis'}
invisible(lapply(r,function(m){
  mod = methyl_psv_results[[m]]
  s = summary(mod,exponentiate = T)$summary_myreg
  print(kable(s,caption = m))
}))
```

#### Annotation

```{r}
metab = sapply(strsplit(r," & "),"[",2)
kable(lipid_anno[lipid_anno$feature_name %in% metab,])

methyl = sapply(strsplit(r," & "),"[",1)
met = anno[anno$Name %in% methyl,c("chr","pos","Name","Probe_rs","Relation_to_Island","UCSC_RefGene_Name")] %>%
  arrange(as.numeric(sub("chr","",chr)),pos)
kable(met)
```

HHAT (Hedgehog Acyltransferase) is a Protein Coding gene. Diseases associated with HHAT include Chondrodysplasia-Pseudohermaphroditism Syndrome and Ancylostomiasis. Among its related pathways are Signaling by GPCR and Notch Signaling Pathway (sino). Gene Ontology (GO) annotations related to this gene include GTP binding and palmitoyltransferase activity. An important paralog of this gene is HHATL.

SH3TC2 (SH3 Domain And Tetratricopeptide Repeats 2) is a Protein Coding gene. Diseases associated with SH3TC2 include Charcot-Marie-Tooth Disease, Type 4C and Mononeuropathy Of The Median Nerve, Mild. An important paralog of this gene is SH3TC1.

EGFL8 (EGF Like Domain Multiple 8) is a Protein Coding gene. Diseases associated with EGFL8 include Ludwig's Angina and Latent Syphilis. Gene Ontology (GO) annotations related to this gene include calcium ion binding. An important paralog of this gene is EGFL7.

TUBGCP3 (Tubulin Gamma Complex Associated Protein 3) is a Protein Coding gene. Diseases associated with TUBGCP3 include Cardiomyopathy, Dilated, 1P. Among its related pathways are G-Beta Gamma Signaling and Regulation of PLK1 Activity at G2/M Transition. Gene Ontology (GO) annotations related to this gene include structural molecule activity and gamma-tubulin binding. An important paralog of this gene is TUBGCP6.

RHOT2 (Ras Homolog Family Member T2) is a Protein Coding gene. Diseases associated with RHOT2 include 3-Methylglutaconic Aciduria, Type Iii and Neuropathy, Congenital Hypomyelinating, 1, Autosomal Recessive. Among its related pathways are Signaling by GPCR and Mitophagy - animal. Gene Ontology (GO) annotations related to this gene include calcium ion binding and GTPase activity. An important paralog of this gene is RHOT1.

WDR90 (WD Repeat Domain 90) is a Protein Coding gene. An important paralog of this gene is EML5. EML5 (EMAP Like 5) is a Protein Coding gene. Diseases associated with EML5 include Mucopolysaccharidosis, Type Ivb. An important paralog of this gene is EML6.

ZNF768 (Zinc Finger Protein 768) is a Protein Coding gene. Among its related pathways are Herpes simplex virus 1 infection. An important paralog of this gene is ZNF287. ZNF287 (Zinc Finger Protein 287) is a Protein Coding gene. Diseases associated with ZNF287 include Nanophthalmos. Among its related pathways are Gene Expression. Gene Ontology (GO) annotations related to this gene include nucleic acid binding and DNA-binding transcription factor activity, RNA polymerase II-specific. An important paralog of this gene is ZNF426.

ATP5PO (ATP Synthase Peripheral Stalk Subunit OSCP) is a Protein Coding gene. Diseases associated with ATP5PO include Hashimoto Thyroiditis and Acute Thyroiditis. Among its related pathways are Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins. and Parkinson disease. An important paralog of this gene is ENSG00000249209.

#### Metabolite at PSV

```{r}
r = lapply(names(metab_psv_results),function(m){
  mod = metab_psv_results[[m]]
  s = summary(mod,exponentiate = T)$summary_myreg
  if (s["tnie","p"] < 0.05){return(m)}else{return(NA)}
})
r = unlist(r[!is.na(r)])
```

There are `r length(r)` pairs with significant NIE:

```{r results='asis'}
invisible(lapply(r,function(m){
  mod = metab_psv_results[[m]]
  s = summary(mod,exponentiate = T)$summary_myreg
  print(kable(s,caption = m))
}))
```

#### Annotation

```{r}
metab = sapply(strsplit(r," & "),"[",2)
kable(lipid_anno[lipid_anno$feature_name %in% metab,])

methyl = sapply(strsplit(r," & "),"[",1)
met = anno[anno$Name %in% methyl,c("chr","pos","Name","Probe_rs","Relation_to_Island","UCSC_RefGene_Name")] %>%
  arrange(as.numeric(sub("chr","",chr)),pos)
kable(met)
```

TBC1D8 (TBC1 Domain Family Member 8) is a Protein Coding gene. Among its related pathways are Cell cycle. Gene Ontology (GO) annotations related to this gene include calcium ion binding and GTPase activator activity. An important paralog of this gene is TBC1D8B.

SLC6A18 (Solute Carrier Family 6 Member 18) is a Protein Coding gene. Diseases associated with SLC6A18 include Iminoglycinuria and Hyperglycinuria. Among its related pathways are Transport of glucose and other sugars, bile salts and organic acids, metal ions and amine compounds and NRF2 pathway. Gene Ontology (GO) annotations related to this gene include neurotransmitter:sodium symporter activity. An important paralog of this gene is SLC6A19.

TCEA2 (Transcription Elongation Factor A2) is a Protein Coding gene. Diseases associated with TCEA2 include Hypotrichosis-Lymphedema-Telangiectasia Syndrome and Uv-Sensitive Syndrome. Among its related pathways are RNA Polymerase II Transcription Initiation And Promoter Clearance. Gene Ontology (GO) annotations related to this gene include nucleic acid binding and translation elongation factor activity. An important paralog of this gene is TCEA1.

# Questions

1. Do we want to include interaction effects? They can be quite difficult to interpret.
2. I have not bootstrapped these CIs. For the selected pairs that we want to write up, are bootstrap CIs better than delta method? Or should I bootstrap everything and use that for significance testing?
