---
title: "Figures for Elizabeth"
author: "Tim Vigers"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: word_document
---

```{r setup, include=FALSE}
library(ggplot2)
library(ggrepel)
library(RColorBrewer)
library(qqman)
knitr::opts_chunk$set(echo = FALSE,message = FALSE,dpi = 600,
                      fig.width = 8,fig.height = 8)
knitr::opts_knit$set(root.dir = "C:/Users/timvigers/Dropbox/School/Statistical Genomics/EWAS/results/late_infancy")
```

```{r}
# Import data
bfdur = read.csv("bfdur_candidates.csv",na.strings = "")
wbr = read.csv("id_wbr_candidates.csv",na.strings = "")
```

```{r plot functions}
manhattan_plot = function(df,p,annotate = NULL,solid_pval = 1e-5,dashed_pval = 5e-8){
  y = sym(p)
  plot = ggplot(df,aes(x = factor(chr),y = -log10(!!y))) + 
    geom_jitter(aes(color = (chr %% 2 == 0))) + 
    xlab("Chromosome") + ylab(expression(-log[10](p))) +
    theme_bw() + scale_color_discrete(guide="none")
  if(!is.null(annotate)){
    df$anno = "No"
    df$anno[df$probe %in% annotate] = "Yes"
    plot = plot + geom_label_repel(data = df[df$anno == "Yes",],aes(label = probe),
                                   segment.colour="black")
  }
  if (!is.null(solid_pval)) {
    plot = plot + geom_hline(yintercept = -log10(solid_pval))
  }
  if (!is.null(dashed_pval)){
    plot = plot + geom_hline(yintercept = -log10(dashed_pval),linetype = "dashed")
  }
  return(plot)
}
qq_plot = function(df,p){
  expected = -log10(ppoints(nrow(df)))
  ps = -log10(sort(df[,p],decreasing = F))
  ggplot(data = NULL, aes(x = expected,y = ps)) + 
    geom_point() + geom_abline(intercept = 0,slope = 1,color = "red") +
    xlab(expression("Expected" -log[10](p))) +
    ylab(expression("Observed" -log[10](p))) +
    theme_bw()
}
```

# Breastfeeding Duration

## Manhattan Plot

```{r}
# Prepare for Manhattan plot
man = bfdur[,c("probe","chr","pos","bfdur.P")]
man$chr = as.numeric(sub("chr","",man$chr))
manhattan_plot(man,p = "bfdur.P", solid_pval = NULL, dashed_pval = NULL,
               annotate = c("cg00574958","cg19693031","cg22369607","cg23307264",
                            "cg24092000","cg27173510"))
# manhattan(man,chr = "chr",bp = "pos",p = "bfdur.P",snp = "probe",
#           highlight = c("cg00574958","cg19693031","cg22369607","cg23307264",
#                         "cg24092000","cg27173510"))
```

## QQ Plot

```{r}
qq_plot(bfdur,"bfdur.P")
```

```{r}
# Calculate genomic inflation factor
chisq = qchisq(bfdur$bfdur.P,1,lower.tail = F)
lambda = median(chisq)/qchisq(0.5,1)
```

# WBR

## < 4 Months

### Manhattan Plot

```{r}
# Prepare for Manhattan plot
man = wbr[,c("probe","chr","pos","id_wbr.4.months.P")]
man$chr = as.numeric(sub("chr","",man$chr))
manhattan_plot(man,p = "id_wbr.4.months.P",solid_pval = NULL, dashed_pval = NULL,
               annotate = "cg26245086")
# manhattan(man,chr = "chr",bp = "pos",p = "id_wbr.4.months.P",snp = "probe",
#           highlight = "cg26245086")
```

### QQ Plot

```{r}
qq_plot(wbr,"id_wbr.4.months.P")
```

## 6+ Months

### Manhattan Plot

```{r}
# Prepare for Manhattan plot
man = wbr[,c("probe","chr","pos","id_wbr6..months.P")]
man$chr = as.numeric(sub("chr","",man$chr))
manhattan_plot(man,p = "id_wbr6..months.P",solid_pval = NULL, dashed_pval = NULL,)
# manhattan(man,chr = "chr",bp = "pos",p = "id_wbr6..months.P",snp = "probe")
```

### QQ Plot

```{r}
qq_plot(wbr,"id_wbr6..months.P")
```
